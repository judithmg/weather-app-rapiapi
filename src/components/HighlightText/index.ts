import styled from 'styled-components';

const color = '#46b96c';

interface Props {
  color?: string;
  size?: number;
}
export default styled.strong<Props>`
  position: relative;
  z-index: 0;
  ${(props) => props.size && `font-size: ${props.size}px`};

  &::after {
    ${(props) =>
      props.color
        ? `background-color: ${props.color}`
        : `background-color: ${color}`};

    content: '';
    position: absolute;
    width: calc(100% + 4px);
    height: 60%;
    left: -2px;
    bottom: 0;
    z-index: -1;
    transform: rotate(-2deg);
  }
`;
