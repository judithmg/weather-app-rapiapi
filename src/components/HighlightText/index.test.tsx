import '@testing-library/jest-dom/extend-expect';

import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';

import HighlightText from '.';

const SAMPLE_TITLE = 'Tis is my HighlightText';

describe('HighlightText', () => {
  it('Renders as expected', () => {
    const tree = renderer
      .create(<HighlightText color="red">{SAMPLE_TITLE}</HighlightText>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('Renders as expect', () => {
    const { getByText } = render(
      <HighlightText color="red">{SAMPLE_TITLE}</HighlightText>,
    );
    expect(getByText(SAMPLE_TITLE)).toBeInTheDocument();
  });
});
