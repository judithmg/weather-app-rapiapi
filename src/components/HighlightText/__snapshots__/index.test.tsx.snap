// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightText renders as expected with state Default 1`] = `
<strong
  className="sc-bczRLJ dqqmOe"
  color="red"
>
  Tis is my HighlightText
</strong>
`;
